src/main/java/Bert/Bert.java

```diff
@@ -0,0 +1,357 @@
+        }
+        String from_DateCheck = cleanLine.substring(cleanLine.indexOf("/from"), cleanLine.indexOf("/to")).trim();
+        String to_DateCheck = cleanLine.substring(cleanLine.indexOf("/to")).trim();
+        if(from_DateCheck.length()<=5 ||  to_DateCheck.length()<=3){
+            throw new EventDateError();
+        }
+    }
+
+    public static void println(String line) {
+        System.out.println(line);
+    }
+    public static void print(String line) {
+        System.out.print(line);
+    }
+    public static void pt(String test) {
+        System.out.println("XX"+test+"XX");
+    }
+
+    private static void writeToFile() throws IOException {
+        String filePath = "./docs/temp.txt";
+        File f = new File(filePath); //create file obj
+        f.createNewFile(); //create file in directory
+        System.out.println("full path: " + f.getAbsolutePath());
+        System.out.println("file exists?: " + f.exists());
+        FileWriter fw = new FileWriter(filePath);
+        for(Task task : taskAL)
+        {
+            fw.write(task.toString());
+            fw.write(System.lineSeparator());
+        }
+        fw.close();
+    }
+    private static void readFromFile() throws IOException {
+        String filePath = "./docs/temp.txt";
+        File f = new File(filePath); // create a File for the given file path
+        if (f.createNewFile()) {           // Try to create the file
+            System.out.println("File created: " + f.getName());
+        } else {
+            println("File already exists.");
+        }
+
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+
+        while (s.hasNext()) {
+            initializeFromTempFile(s);
+        }
+        if(!taskAL.isEmpty()){
+            println("Tasks have been initialized.");
+            listTasks();
+        }
+    }
+
+    private static void initializeFromTempFile(Scanner s) {
+        String line = s.nextLine();
+        String taskType = line.substring(1,2);
+        String description = line.substring(7);
+
+        if(taskType.equalsIgnoreCase("T")){
+            taskAL.add(new Todo(description));
+        }
+        else if(taskType.equalsIgnoreCase("D")){
+            String taskName = description.substring(0,description.indexOf("(by:")).trim();
+            String byDate =  description.substring(description.indexOf("(by:")+4).trim();
+            byDate = byDate.replace(")","");
+            taskAL.add(new Deadline(taskName, byDate));
+        } else if(taskType.equalsIgnoreCase("E")){
+            String taskName = description.substring(0,description.indexOf("(From:")).trim();
+            String fromTime =  description.substring(description.indexOf("(From:")+6,description.indexOf("--")).trim();
+            String toTime =  description.substring(description.indexOf("To:")+3).trim();
+            toTime = toTime.replace(")","");
+            taskAL.add(new Event(taskName, fromTime, toTime));
+        }
+    }
+}
```

Comment:
Your main class is doing prettymuch most of the heavy lifting in the codebase.
Try to extract more classes for better code organization before the final submission